--- Pike-v7.6.86/src/modules/Image/configure.in.orig	2006-04-10 22:48:34.000000000 +0200
+++ Pike-v7.6.86/src/modules/Image/configure.in	2006-12-23 15:16:10.503925299 +0100
@@ -27,22 +27,17 @@
       # Darwin x86 can output ELF even though we can't use it
       # FIXME: Investigate using nasm -f macho for Mach-O output
       AC_MSG_RESULT([no, Mach-O not yet supported])
-    elif test "`uname -m`" = "ia64"; then
-      # nasm/ia64 can compile ia32 code...
-      AC_MSG_RESULT([yes, but we are on an IA64 machine])
-    elif test "`uname -m`" = "x86_64"; then
-      # nasm/x86_64 can also compile ia32 code.
-      # It can even link with it, but it's not a good idea...
-      AC_MSG_RESULT([yes, but we are on an x86_64 machine])
-    elif test "`uname -s`" = "OpenBSD"; then
-      #nasm on OpenBSD is braindamaged
-      AC_MSG_RESULT(no)
-    elif test x"${with_valgrind:-no}" != xno; then
-      AC_MSG_RESULT([no - valgrind is used])
     else
-      ASSEMBLY_OBJECTS=assembly.o
-      AC_DEFINE(ASSEMBLY_OK)
-      AC_MSG_RESULT(yes)
+      case "$build_alias" in
+	i?86-*|athlon-*)
+	  ASSEMBLY_OBJECTS=assembly.o
+	  AC_DEFINE(ASSEMBLY_OK)
+	  AC_MSG_RESULT(yes)
+	  ;;
+	*)
+	  AC_MSG_RESULT([yes, but we are doing non-x86 build])
+	  ;;
+      esac
     fi
   else
     AC_MSG_RESULT(no)
