--- Pike-v7.4.20/src/aclocal.m4.orig	Thu Apr  3 18:58:36 2003
+++ Pike-v7.4.20/src/aclocal.m4	Fri Apr 25 13:52:39 2003
@@ -4,6 +4,7 @@
 dnl newer autoconf call substr m4_substr
 ifdef([substr], ,m4_copy(m4_substr,substr))
 
+define([AC_ACVERSION], m4_PACKAGE_VERSION)
 
 pushdef([AC_PROG_CC_WORKS],
 [
@@ -156,19 +157,19 @@
   ORIG_AC_CHECK_SIZEOF($1,$2)
 ])
 
-define([ORIG_CHECK_HEADERS], defn([AC_CHECK_HEADERS]))
-pushdef([AC_CHECK_HEADERS],
-[
-  if test "x$enable_binary" != "xno"; then
-    ORIG_CHECK_HEADERS($1,$2,$3)
-  else
-    for ac_hdr in $1
-    do
-      ac_safe=`echo "$ac_hdr" | sed 'y%./+-%__p_%'`
-      eval "ac_cv_header_$ac_safe=yes"
-    done
-  fi
-])
+dnl define([ORIG_CHECK_HEADERS], defn([AC_CHECK_HEADERS]))
+dnl pushdef([AC_CHECK_HEADERS],
+dnl [
+dnl   if test "x$enable_binary" != "xno"; then
+dnl     ORIG_CHECK_HEADERS($1,$2,$3)
+dnl   else
+dnl     for ac_hdr in $1
+dnl     do
+dnl       ac_safe=`echo "$ac_hdr" | sed 'y%./+-%__p_%'`
+dnl       eval "ac_cv_header_$ac_safe=yes"
+dnl     done
+dnl   fi
+dnl ])
 
 AC_DEFUN(AC_MY_CHECK_TYPE,
 [
--- Pike-v7.4.20/src/configure.in.orig	Fri Apr 25 13:41:36 2003
+++ Pike-v7.4.20/src/configure.in	Fri Apr 25 13:53:48 2003
@@ -2980,9 +2980,6 @@
 ########################################################################
 
 # NOTE: The following test is hardcoded to fail when crosscompiling
-# It also clobbers $LIBOBJS.
-
-OLD_LIBOBJS="${LIBOBJS}"
 
 AC_FUNC_MEMCMP
 
@@ -2990,8 +2987,6 @@
   AC_DEFINE(HAVE_MEMCMP)
 fi
 
-LIBOBJS="${OLD_LIBOBJS}"
-
 ########################################################################
 
 AC_FUNC_MMAP
