diff -urN Pike-v7.6.24.org/src/Makefile.in Pike-v7.6.24/src/Makefile.in
--- Pike-v7.6.24.org/src/Makefile.in	2004-04-23 19:18:34.000000000 +0200
+++ Pike-v7.6.24/src/Makefile.in	2005-02-19 17:56:20.775640896 +0100
@@ -42,17 +42,17 @@
 buildroot=
 
 prefix = @prefix@
-exec_prefix = $(prefix)/bin
-lib_prefix = $(prefix)/lib/pike
+exec_prefix = @bindir@
+lib_prefix = @libdir@/pike
 data_prefix = @datadir@
 share_prefix = $(data_prefix)/pike
-man_prefix = $(prefix)/man
+man_prefix = @mandir@
 
 # Arguments for test_pike.pike used by the verify targets.
 # TESTARGS = -F -a
 
 # Arguments for install.pike used by the install targets.
-INSTALLARGS = --new-style
+INSTALLARGS = --traditional
 
 #
 # use bison please, yacc isn't good enough.
